lib/DXIL/DxilMetadataHelper.cpp

```diff
@@ -284,13 +284,20 @@ void DxilMDHelper::LoadDxilShaderModel(const ShaderModel *&pSM) {
       "_" + std::to_string(Major) + "_" +
       (Minor == ShaderModel::kOfflineMinor ? "x" : std::to_string(Minor));
   pSM = ShaderModel::GetByName(ShaderModelName.c_str());
-  if (!pSM->IsValidForDxil()) {
-    char ErrorMsgTxt[40];
-    StringCchPrintfA(ErrorMsgTxt, _countof(ErrorMsgTxt),
-                     "Unknown shader model '%s'", ShaderModelName.c_str());
-    string ErrorMsg(ErrorMsgTxt);
-    throw hlsl::Exception(DXC_E_INCORRECT_DXIL_METADATA, ErrorMsg);
+  // check to see if the provided shader model is the pre-release version
+  if (ShaderModel::IsPreReleaseShaderModel(Major, Minor)) {
```

Comment:
Why is this change necessary?  I don't think any special case is necessary here for preview shader models.